apps/condo/migrations/20210803143609-0033_alter_ticket_classifier.js

```diff
@@ -0,0 +1,30 @@
+// auto generated by kmigrator
+// KMIGRATOR:0033_alter_ticket_classifier:IyBHZW5lcmF0ZWQgYnkgRGphbmdvIDMuMi41IG9uIDIwMjEtMDgtMDMgMDk6MzYKCmZyb20gZGphbmdvLmRiIGltcG9ydCBtaWdyYXRpb25zLCBtb2RlbHMKaW1wb3J0IGRqYW5nby5kYi5tb2RlbHMuZGVsZXRpb24KCgpjbGFzcyBNaWdyYXRpb24obWlncmF0aW9ucy5NaWdyYXRpb24pOgoKICAgIGRlcGVuZGVuY2llcyA9IFsKICAgICAgICAoJ19kamFuZ29fc2NoZW1hJywgJzAwMzJfYWx0ZXJfZm9yZ290cGFzc3dvcmRhY3Rpb25fdG9rZW4nKSwKICAgIF0KCiAgICBvcGVyYXRpb25zID0gWwogICAgICAgIG1pZ3JhdGlvbnMuQWx0ZXJGaWVsZCgKICAgICAgICAgICAgbW9kZWxfbmFtZT0ndGlja2V0JywKICAgICAgICAgICAgbmFtZT0nY2xhc3NpZmllcicsCiAgICAgICAgICAgIGZpZWxkPW1vZGVscy5Gb3JlaWduS2V5KGJsYW5rPVRydWUsIGRiX2NvbHVtbj0nY2xhc3NpZmllcicsIG51bGw9VHJ1ZSwgb25fZGVsZXRlPWRqYW5nby5kYi5tb2RlbHMuZGVsZXRpb24uU0VUX05VTEwsIHJlbGF0ZWRfbmFtZT0nKycsIHRvPSdfZGphbmdvX3NjaGVtYS50aWNrZXRjbGFzc2lmaWVyJyksCiAgICAgICAgKSwKICAgIF0K
+
+exports.up = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Alter field classifier on ticket
+--
+SET CONSTRAINTS "Ticket_classifier_7a786997_fk_TicketClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_classifier_7a786997_fk_TicketClassifier_id";
+ALTER TABLE "Ticket" ALTER COLUMN "classifier" DROP NOT NULL;
+ALTER TABLE "Ticket" ADD CONSTRAINT "Ticket_classifier_7a786997_fk_TicketClassifier_id" FOREIGN KEY ("classifier") REFERENCES "TicketClassifier" ("id") DEFERRABLE INITIALLY DEFERRED;
+COMMIT;
+
+    `)
+}
+
+exports.down = async (knex) => {
+    await knex.raw(`
+    BEGIN;
+--
+-- Alter field classifier on ticket
+--
+SET CONSTRAINTS "Ticket_classifier_7a786997_fk_TicketClassifier_id" IMMEDIATE; ALTER TABLE "Ticket" DROP CONSTRAINT "Ticket_classifier_7a786997_fk_TicketClassifier_id";
+ALTER TABLE "Ticket" ALTER COLUMN "classifier" SET NOT NULL;
```

Comment:
Setting back `NOT NULL` will not work, if we will rollback this migration after tickets without classifier will be created. We need to set something to tickets without assigned classifier. It can be "Other" classifier.
@pahaz, is it ok?


```suggestion
UPDATE "Ticket" SET classifier = '2e5e5950-f695-4948-a388-4430f4790305' WHERE classifier IS NULL;
ALTER TABLE "Ticket" ALTER COLUMN "classifier" SET NOT NULL;
```

Reply:
I think we can leave it as is or set it as "other".

Reply:
do it in d1f7075